src/core/meshcop/joiner_router.cpp

```diff
@@ -91,7 +91,7 @@ void JoinerRouter::HandleNetifStateChanged(uint32_t aFlags)
 
     mNetif.GetIp6Filter().RemoveUnsecurePort(mSocket.GetSockName().mPort);
 
-    if (mNetif.GetNetworkDataLeader().GetCommissioningData() != NULL)
+    if (mNetif.GetNetworkDataLeader().GetCommissioningDataSubTlv(MeshCoP::Tlv::kBorderAgentLocator) != NULL)
```

Comment:
This should use the new `LeaderBase:: IsJoiningEnabled()` method coming with #1665.